File: Leetcode/typescript/sort-array.ts
function sortArray(nums: number[]): number[] {
  // i basically just made merge sort, nothing special

  function mergeSort(arr: number[]) {
    if (arr.length <= 1) {
      return arr;
    }

    const mid = Math.floor(arr.length / 2);

    const leftMerge = mergeSort(arr.slice(0, mid));
    const rightMerge = mergeSort(arr.slice(mid));
    return merge(leftMerge, rightMerge);
  }
  function merge(left, right) {
    let sortedList: number[] = [];
    let leftIndex = 0;
    let rightIndex = 0;

    while (leftIndex < left.length && rightIndex < right.length) {
      if (left[leftIndex] < right[rightIndex]) {
        sortedList.push(left[leftIndex]);
        leftIndex++;
      } else {
        sortedList.push(right[rightIndex]);
        rightIndex++;
      }
    }

    while (leftIndex < left.length) {
      sortedList.push(left[leftIndex]);
      leftIndex++;
    }
    while (rightIndex < right.length) {
      sortedList.push(right[rightIndex]);
      rightIndex++;
    }
    return sortedList;
  }
  return mergeSort(nums);
}

File: Leetcode/python/sort-array.py
from typing import List

class Solution:
    def sortArray(self, nums: List[int]) -> List[int]:

        def mergeSort(arr):
            if len(arr) <= 1:
                return arr

            mid = len(arr) // 2

            leftMerge = mergeSort(arr[:mid])
            rightMerge = mergeSort(arr[mid:])
            return merge(leftMerge, rightMerge)

        def merge(left, right):
            sortedList = []
            leftIndex, rightIndex = 0, 0

            while leftIndex < len(left) and rightIndex < len(right):
                if left[leftIndex] < right[rightIndex]:
                    sortedList.append(left[leftIndex])
                    leftIndex += 1
                else:
                    sortedList.append(right[rightIndex])
                    rightIndex += 1

            while leftIndex < len(left):
                sortedList.append(left[leftIndex])
                leftIndex += 1
            while rightIndex < len(right):
                sortedList.append(right[rightIndex])
                rightIndex += 1
            return sortedList

        return mergeSort(nums)
